wgsl/index.bs

```diff
@@ -4483,6 +4498,34 @@ User-defined IO must not be passed to [=compute=] shader entry points.
 User-defined IO must be [=numeric scalar=] or [=numeric vector=] types .
 All user defined IO must be assigned locations (See [[#input-output-locations]]).
 
+#### Interpolation #### {#interpolation}
+
+Authors can control how user-defined IO data is interpolated through the use of
+the [=attribute/interpolate=] attribute.
+[SHORTNAME] offers two aspects of interpolation to control: the type of
+interpolation, and the sampling of the interpolation.
+
+The <dfn noexport>interpolation type</dfn> must be one of:
+* `perspective` - Values are interpolated in a perspective correct manner.
+* `linear` - Values are interpolated in a linear, non-perspective correct manner.
+* `flat` - Values are not interpolated.
+    Interpolation sampling is not used with `flat` interpolation.
+
+The <dfn noexport>interpolation sampling</dfn> must be one of:
+* `center` - Interpolation is performed at the center of the pixel.
+* `centroid` - Interpolation is performed within both the fragment and samples
```

Comment:
could this be reworded into something like "interpolation is performed at the average point between the samples covered by the primitive"?

Reply:
I think that's overstating the behaviour.

Vulkan:

https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#shaders-interpolation-decorations

> If a fragment shader input is decorated with Centroid, a single value **may** be assigned to that variable for all samples in the fragment, but that value **must** be interpolated to a location that lies in both the fragment and in the primitive being rendered, including any of the fragment’s samples covered by the primitive.

Reply:
Ok, that's still more clear than what the PR has.
How about: "interpolation is performed at some point within the fragment that is covered by the primitive"?